src/test/RippleKey_test.cpp

```diff
@@ -0,0 +1,358 @@
+        auto const seed = generateSeed(passphrase);
+        auto const secretKey = generateKeyPair(kt, seed).second;
+
+        // Make sure there are no extra fields
+        BEAST_EXPECT(jKeys.size() == 9);
+        BEAST_EXPECT(jKeys[jss::account_id] ==
+            toBase58(calcAccountID(key.publicKey())));
+        BEAST_EXPECT(jKeys[jss::key_type] == to_string(kt));
+        BEAST_EXPECT(jKeys[jss::master_key] == seedAs1751(seed));
+        BEAST_EXPECT(jKeys[jss::master_seed] == toBase58(seed));
+        BEAST_EXPECT(jKeys[jss::master_seed_hex] ==
+            strHex(seed.data(), seed.size()));
+        BEAST_EXPECT(jKeys[jss::public_key] ==
+            toBase58(TOKEN_ACCOUNT_PUBLIC, key.publicKey()));
+        BEAST_EXPECT(jKeys[jss::public_key_hex] ==
+            strHex(key.publicKey().data(), key.publicKey().size()));
+        BEAST_EXPECT(jKeys["secret_key"] ==
+            toBase58(TOKEN_ACCOUNT_SECRET, secretKey));
+        BEAST_EXPECT(jKeys["secret_key_hex"] ==
+            strHex(secretKey.data(), secretKey.size()));
+    }
+
+    void
+    testSign(ripple::KeyType const kt)
+    {
+        using namespace serialize;
+        using namespace ripple;
+
+        auto obj = deserialize(getKnownTxSigned().SerializedText);
+
+        if (!BEAST_EXPECT(obj))
+            return;
+
+        ripple::STTx tx{ std::move(*obj) };
+        // The hard-coded version is signed
+        auto check = tx.checkSign(true);
+        BEAST_EXPECT(check.first);
+        {
+            auto const key = RippleKey::make_RippleKey(kt,
+                std::string("alice"));
+            auto const expectedSignature = kt == KeyType::secp256k1 ?
+                tx.getFieldVL(sfTxnSignature) :
+                strUnHex("0751E8D38C26E8B6C953766A8A58570CA0CB93E57B86047F1FEF8DA3D7"
+                    "9DFB97E78F4E59365C88EEE0E94EF7C1A2155A828B239AC00F3E95802D"
+                    "851ABB113F06").first;
+            auto const expectedSigningKey = kt == KeyType::secp256k1 ?
+                tx.getFieldVL(sfSigningPubKey) :
+                strUnHex("ED4A9D72F2557B714713DC8BA7C6F9576BCC06117A52F6C32"
+                    "F1E26FEEF9819EC8E").first;
+
+            // Remove the signature
+            tx.makeFieldAbsent(sfTxnSignature);
+            tx.setAccountID(sfAccount, calcAccountID(key.publicKey()));
+            check = tx.checkSign(true);
+            BEAST_EXPECT(!check.first);
+            BEAST_EXPECT(check.second == "Invalid signature.");
+
+            // Now re-sign it
+            key.singleSign(tx);
+            BEAST_EXPECT(tx.checkSign(true).first);
+            // Same signature
+            BEAST_EXPECT(tx.getFieldVL(sfSigningPubKey) == expectedSigningKey);
+            BEAST_EXPECT(tx.getFieldVL(sfTxnSignature) == expectedSignature);
+            BEAST_EXPECT(!tx.isFieldPresent(sfSigners));
+        }
+
+        {
+            // Use a different key to multisign, because an account can't
+            // multisign its own transaction.
```

Comment:
I did test that by accident earlier, but changed the signers. That kind of test is more of a test of the underlying ripple / ripple-libpp code, and should (and is AFAIK) tested there. I don't think it's really appropriate as a test in this app.